parseAPI/src/Block.C

```diff
@@ -356,3 +356,28 @@ void Block::moveTargetEdges(Block* B) {
 	}
     trgs.clear();
 }
+
+void Block::copy_sources(edgelist & src) {
+    boost::lock_guard<Block> g(*this);
+    src = _srclist;
+}
+
+void Block::copy_targets(edgelist & trg) {
+    boost::lock_guard<Block> g(*this);
+    trg = _trglist;
+}
+
+bool Block::hasCallSource() {
+    boost::lock_guard<Block> g(*this);
+    for (auto e: _srclist)
+        if (e->type() == CALL)
+            return true;
+    return false;
+}
+
+bool Block::hasIncomingOtherThan(Edge* e) {
+    boost::lock_guard<Block> g(*this);
+    return _srclist.empty() ? false :
+            _srclist.size() > 1 ? true :
+            *_srclist.begin() != e;
+}
```

Comment:
I've replaced this logic with the simpler `getOnlyIncomingEdge`, which returns the only element of `_srclist` if it only has one element (and `nullptr` otherwise).

This does change an edge case in Parser.C, now if `trg_block` has no incoming edges it won't mark `e` as a non-tail call, previously it did. I presume (haven't tested) that this should never happen, but if it's an issue I can adjust again.

Reply:
Should be fine. If this case happens, it means an actual incoming edge does not show up in the source edge list, this means a broken CFG.